libraries/fabric-shim/test/unit/utils/statebased.js

```diff
@@ -106,33 +110,70 @@ describe('KeyEndorsementPolicy', () => {
             const policy = ep.getPolicy();
             const anotherEp = new KeyEndorsementPolicy(policy);
 
-            const spe = common.SignaturePolicyEnvelope.deserializeBinary(policy);
-            const speClone = common.SignaturePolicyEnvelope.deserializeBinary(anotherEp.getPolicy());
+            const spe =
+                common.SignaturePolicyEnvelope.deserializeBinary(policy);
+            const speClone = common.SignaturePolicyEnvelope.deserializeBinary(
+                anotherEp.getPolicy()
+            );
             expect(spe.toObject()).to.deep.equals(speClone.toObject());
         });
 
-
         it('should get policy that is semantically valid', () => {
             const policy = ep.getPolicy();
-            const spe = common.SignaturePolicyEnvelope.deserializeBinary(policy);
+            const spe =
+                common.SignaturePolicyEnvelope.deserializeBinary(policy);
 
             // create a blank object and expand all the protobufs into it
             const speObject = spe.toObject();
 
-            speObject.identitiesList = spe.getIdentitiesList().map(principal => {
-                let mapped = { principalClassification: 0 };
-                mapped.principal = msp.MSPRole.deserializeBinary(principal.getPrincipal_asU8()).toObject();
-                return mapped;
-            });            
-
-            speObject.rule.nOutOf.rulesList = spe.getRule().getNOutOf().getRulesList().map(sigRule =>{
-                 return {signedBy: sigRule.getSignedBy()}
-            });
-
-            const expectedPolicy={"version":0,"rule":{"signedBy":0,"nOutOf":{"n":3,"rulesList":[{"signedBy":0},{"signedBy":1},{"signedBy":2}]}},"identitiesList":[{"principalClassification":0,"principal":{"mspIdentifier":"Org1MSP","role":0}},{"principalClassification":0,"principal":{"mspIdentifier":"Org2MSP","role":0}},{"principalClassification":0,"principal":{"mspIdentifier":"Org3MSP","role":0}}]}
+            speObject.identitiesList = spe
+                .getIdentitiesList()
+                .map((principal) => {
+                    let mapped = { principalClassification: 0 };
+                    mapped.principal = msp.MSPRole.deserializeBinary(
+                        principal.getPrincipal_asU8()
+                    ).toObject();
+                    return mapped;
+                });
+
+            speObject.rule.signedBy = spe.getRule().getSignedBy();
```

Comment:
This line is the only actual change in this file. The rest is just formatting.

It seems newer binding defaulted the value of `speObject.rule.signedBy` to `undefined` instead of `0` when created by `spe.toObject()` in the test code above. This change just explicitly copies the actual value instead of relying on a default, and is probably how it should always have been done.

Note that this is just a change to how the test code collects the protobuf state before the equality test below. No change to the implementation code.